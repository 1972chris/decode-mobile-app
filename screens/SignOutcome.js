import React from 'react';
import { Text, View, Linking } from 'react-native';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import ImageOverlay from 'react-native-image-overlay';
import { translate } from 'react-i18next';
import openPetitionInBrowser from '../application/utils';
import styles from './styles';
import Button from '../application/components/Button/Button';
import Error from './Error';
import i18n from '../i18n';

const successImage = require('../assets/images/city.png');

const linksStyle = {
  textAlign: 'left',
  color: '#00F',
  fontSize: 16,
  fontWeight: 'bold',
  textDecorationLine: 'underline',
  marginVertical: 10,
};

class SignOutcome extends React.Component {
  static route = {
    navigationBar: {
      backgroundColor: 'white',
      tintColor: 'rgb(0,163,158)',
      title: 'Outcome',
    },
  }

  static handlePress() {
    Linking.openURL('http://secure-petitions.s3-website-eu-west-1.amazonaws.com/#/results/59f888c8ce33c76884e8cf16');
  }

  static createLink(linkText, index) {
    return (
      <Text
        key={index}
        style={linksStyle}
        onPress={() => { alert('End MVP'); }} // eslint-disable-line
      >
        {linkText}
      </Text>);
  }

  successful() {
    const links = this.props.links.map(SignOutcome.createLink);

    return (
      <View style={styles.signOutcomeContainer}>
        <ImageOverlay
          source={successImage}
          overlayAlpha={0}
          contentPosition="bottom"
        >
          <Text
            style={{
              fontSize: 34,
              color: 'white',
              marginBottom: 30,
              marginLeft: 20,
              marginRight: 20,
              textAlign: 'center',
              backgroundColor: 'transparent',
            }}
          >
            { this.props.t('voteRecorded')}
          </Text>
        </ImageOverlay>
        <View style={{
          marginTop: 30,
          flex: 1,
        }}
        >
          <Text style={{
            marginBottom: 20,
            fontSize: 16,
          }}
          >
            {this.props.t('maybeInterested')}
          </Text>

          {links}
        </View>
        <Button
          name={this.props.t('backDecidim')}
          onPress={() => openPetitionInBrowser(this.props.petition.id)}
          style={{
            width: 200,
            alignSelf: 'center',
          }}
        />
      </View>
    );
  }

  render() {
    if (!this.props.signSuccess) {
      return <Error title={this.props.petition.title} />;
    }
    return this.successful();
  }
}

SignOutcome.propTypes = {
  signSuccess: PropTypes.bool.isRequired,
  petition: PropTypes.shape({
    title: PropTypes.string,
    description: PropTypes.string,
    closingDate: PropTypes.string,
    id: PropTypes.string,
  }),
  links: PropTypes.arrayOf(PropTypes.string),
  t: PropTypes.func.isRequired,
};

SignOutcome.defaultProps = {
  petition: undefined,
  links: [
    'Make Poblenou a car-free zone',
    'Reduce noise pollution'],
};

const mapStateToProps = state => ({
  signSuccess: state.signOutcome.signSuccess,
  petition: state.petition.petition,
});

const mapDispatchToProps = () => ({ });

export default translate('signOutcome', { i18n })(connect(mapStateToProps, mapDispatchToProps)(SignOutcome));
