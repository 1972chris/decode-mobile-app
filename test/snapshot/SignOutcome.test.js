import React from 'react';
import configureStore from 'redux-mock-store';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import thunk from 'redux-thunk';
import SignOutcome from '../../screens/SignOutcome';

Enzyme.configure({ adapter: new Adapter() });

const mockStore = configureStore([thunk]);

const initialState = {
  decidimInfo: {
    decidimAPIUrl: 'http://something.com',
  },
  signOutcome: {
    signSuccess: false,
  },
  petition: {
    petition: {},
    error: '',
  },
};

it('renders signOutcome component on successful sign', () => {
  const state = {
    ...initialState,
    signOutcome: {
      signSuccess: true,
    },
  };
  const wrapper = shallow(
    <SignOutcome />,
    { context: { store: mockStore(state) } },
  );
  expect(wrapper.dive()).toMatchSnapshot();
});

it('renders signOutcome component on unsuccessful sign', () => {
  const state = {
    ...initialState,
    signOutcome: {
      signSuccess: false,
    },
  };
  const wrapper = shallow(
    <SignOutcome />,
    { context: { store: mockStore(state) } },
  );
  expect(wrapper.dive()).toMatchSnapshot();
});
