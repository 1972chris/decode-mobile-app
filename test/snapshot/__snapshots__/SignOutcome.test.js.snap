// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders signOutcome component on successful sign 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignOutcome
    goToAuthorization={[Function]}
    petition={Object {}}
    petitionError=""
    petitionLink="http://something.com"
    signSuccess={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Image
          source={1}
          style={
            Object {
              "alignSelf": "center",
              "height": 75,
              "marginBottom": 40,
              "width": 75,
            }
          }
        />,
        <Button
          enabled={true}
          name="Back to home2"
          onPress={[Function]}
          style={Object {}}
        />,
      ],
      "style": Object {
        "alignItems": "center",
        "backgroundColor": "rgb(246, 246, 246)",
        "flex": 1,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "source": 1,
          "style": Object {
            "alignSelf": "center",
            "height": 75,
            "marginBottom": 40,
            "width": 75,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "enabled": true,
          "name": "Back to home2",
          "onPress": [Function],
          "style": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Image
            source={1}
            style={
              Object {
                "alignSelf": "center",
                "height": 75,
                "marginBottom": 40,
                "width": 75,
              }
            }
          />,
          <Button
            enabled={true}
            name="Back to home2"
            onPress={[Function]}
            style={Object {}}
          />,
        ],
        "style": Object {
          "alignItems": "center",
          "backgroundColor": "rgb(246, 246, 246)",
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "source": 1,
            "style": Object {
              "alignSelf": "center",
              "height": 75,
              "marginBottom": 40,
              "width": 75,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "enabled": true,
            "name": "Back to home2",
            "onPress": [Function],
            "style": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;

exports[`renders signOutcome component on unsuccessful sign 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignOutcome
    goToAuthorization={[Function]}
    petition={Object {}}
    petitionError=""
    petitionLink="http://something.com"
    signSuccess={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <View
        style={
          Object {
            "alignSelf": "stretch",
            "backgroundColor": "#FFF",
            "margin": 16,
            "marginTop": 30,
            "paddingHorizontal": 20,
            "paddingVertical": 40,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "alignSelf": "center",
              "height": 75,
              "marginBottom": 40,
              "width": 75,
            }
          }
        />
        <View
          style={
            Object {
              "alignSelf": "center",
              "width": 250,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 14,
                "marginBottom": 4,
                "textAlign": "center",
              }
            }
          >
            Sign failed for
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 20,
                "fontWeight": "500",
                "marginVertical": 15,
                "textAlign": "center",
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "marginBottom": 20,
                "textAlign": "center",
              }
            }
          >
            Reason:  

You can return to view other petitions on the Secure Petitions website
          </Text>
          <Button
            enabled={true}
            name="Back to home"
            onPress={[Function]}
            style={Object {}}
          />
        </View>
      </View>,
      "style": Object {
        "alignItems": "center",
        "backgroundColor": "rgb(246, 246, 246)",
        "flex": 1,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Image
            source={1}
            style={
              Object {
                "alignSelf": "center",
                "height": 75,
                "marginBottom": 40,
                "width": 75,
              }
            }
          />,
          <View
            style={
              Object {
                "alignSelf": "center",
                "width": 250,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 14,
                  "marginBottom": 4,
                  "textAlign": "center",
                }
              }
            >
              Sign failed for
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "500",
                  "marginVertical": 15,
                  "textAlign": "center",
                }
              }
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "marginBottom": 20,
                  "textAlign": "center",
                }
              }
            >
              Reason:  

You can return to view other petitions on the Secure Petitions website
            </Text>
            <Button
              enabled={true}
              name="Back to home"
              onPress={[Function]}
              style={Object {}}
            />
          </View>,
        ],
        "style": Object {
          "alignSelf": "stretch",
          "backgroundColor": "#FFF",
          "margin": 16,
          "marginTop": 30,
          "paddingHorizontal": 20,
          "paddingVertical": 40,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "source": 1,
            "style": Object {
              "alignSelf": "center",
              "height": 75,
              "marginBottom": 40,
              "width": 75,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 14,
                    "marginBottom": 4,
                    "textAlign": "center",
                  }
                }
              >
                Sign failed for
              </Text>,
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 20,
                    "fontWeight": "500",
                    "marginVertical": 15,
                    "textAlign": "center",
                  }
                }
              />,
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "marginBottom": 20,
                    "textAlign": "center",
                  }
                }
              >
                Reason:  

You can return to view other petitions on the Secure Petitions website
              </Text>,
              <Button
                enabled={true}
                name="Back to home"
                onPress={[Function]}
                style={Object {}}
              />,
            ],
            "style": Object {
              "alignSelf": "center",
              "width": 250,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "Sign failed for",
                "ellipsizeMode": "tail",
                "style": Object {
                  "fontSize": 14,
                  "marginBottom": 4,
                  "textAlign": "center",
                },
              },
              "ref": null,
              "rendered": "Sign failed for",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": undefined,
                "ellipsizeMode": "tail",
                "style": Object {
                  "fontSize": 20,
                  "fontWeight": "500",
                  "marginVertical": 15,
                  "textAlign": "center",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "Reason:  

You can return to view other petitions on the Secure Petitions website",
                "ellipsizeMode": "tail",
                "style": Object {
                  "marginBottom": 20,
                  "textAlign": "center",
                },
              },
              "ref": null,
              "rendered": "Reason:  

You can return to view other petitions on the Secure Petitions website",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "enabled": true,
                "name": "Back to home",
                "onPress": [Function],
                "style": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <View
          style={
            Object {
              "alignSelf": "stretch",
              "backgroundColor": "#FFF",
              "margin": 16,
              "marginTop": 30,
              "paddingHorizontal": 20,
              "paddingVertical": 40,
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "alignSelf": "center",
                "height": 75,
                "marginBottom": 40,
                "width": 75,
              }
            }
          />
          <View
            style={
              Object {
                "alignSelf": "center",
                "width": 250,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 14,
                  "marginBottom": 4,
                  "textAlign": "center",
                }
              }
            >
              Sign failed for
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "500",
                  "marginVertical": 15,
                  "textAlign": "center",
                }
              }
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "marginBottom": 20,
                  "textAlign": "center",
                }
              }
            >
              Reason:  

You can return to view other petitions on the Secure Petitions website
            </Text>
            <Button
              enabled={true}
              name="Back to home"
              onPress={[Function]}
              style={Object {}}
            />
          </View>
        </View>,
        "style": Object {
          "alignItems": "center",
          "backgroundColor": "rgb(246, 246, 246)",
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Image
              source={1}
              style={
                Object {
                  "alignSelf": "center",
                  "height": 75,
                  "marginBottom": 40,
                  "width": 75,
                }
              }
            />,
            <View
              style={
                Object {
                  "alignSelf": "center",
                  "width": 250,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 14,
                    "marginBottom": 4,
                    "textAlign": "center",
                  }
                }
              >
                Sign failed for
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 20,
                    "fontWeight": "500",
                    "marginVertical": 15,
                    "textAlign": "center",
                  }
                }
              />
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "marginBottom": 20,
                    "textAlign": "center",
                  }
                }
              >
                Reason:  

You can return to view other petitions on the Secure Petitions website
              </Text>
              <Button
                enabled={true}
                name="Back to home"
                onPress={[Function]}
                style={Object {}}
              />
            </View>,
          ],
          "style": Object {
            "alignSelf": "stretch",
            "backgroundColor": "#FFF",
            "margin": 16,
            "marginTop": 30,
            "paddingHorizontal": 20,
            "paddingVertical": 40,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "source": 1,
              "style": Object {
                "alignSelf": "center",
                "height": 75,
                "marginBottom": 40,
                "width": 75,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "fontSize": 14,
                      "marginBottom": 4,
                      "textAlign": "center",
                    }
                  }
                >
                  Sign failed for
                </Text>,
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "fontSize": 20,
                      "fontWeight": "500",
                      "marginVertical": 15,
                      "textAlign": "center",
                    }
                  }
                />,
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "marginBottom": 20,
                      "textAlign": "center",
                    }
                  }
                >
                  Reason:  

You can return to view other petitions on the Secure Petitions website
                </Text>,
                <Button
                  enabled={true}
                  name="Back to home"
                  onPress={[Function]}
                  style={Object {}}
                />,
              ],
              "style": Object {
                "alignSelf": "center",
                "width": 250,
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "accessible": true,
                  "allowFontScaling": true,
                  "children": "Sign failed for",
                  "ellipsizeMode": "tail",
                  "style": Object {
                    "fontSize": 14,
                    "marginBottom": 4,
                    "textAlign": "center",
                  },
                },
                "ref": null,
                "rendered": "Sign failed for",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "accessible": true,
                  "allowFontScaling": true,
                  "children": undefined,
                  "ellipsizeMode": "tail",
                  "style": Object {
                    "fontSize": 20,
                    "fontWeight": "500",
                    "marginVertical": 15,
                    "textAlign": "center",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "accessible": true,
                  "allowFontScaling": true,
                  "children": "Reason:  

You can return to view other petitions on the Secure Petitions website",
                  "ellipsizeMode": "tail",
                  "style": Object {
                    "marginBottom": 20,
                    "textAlign": "center",
                  },
                },
                "ref": null,
                "rendered": "Reason:  

You can return to view other petitions on the Secure Petitions website",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "enabled": true,
                  "name": "Back to home",
                  "onPress": [Function],
                  "style": Object {},
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
